pad_ui/PackDownloadDiv.gd

```diff
@@ -0,0 +1,149 @@
+	var pack_installed = pad_manager.get_pack_location(pack_name) != null
+	if pack_installed:
+		_show_completed_ui()
+	else:
+		_show_init_ui()
+
+func _convert_byte_to_stepified_megabytes(byte_value : int):
+	# convert bytes to MB with 2 digits of precision
+	return stepify(byte_value / 1000000, 0.01)
+
+func _process(delta):
+	if request_obj == null:
+		return
+
+	var should_update_ui = request_obj != null and request_obj.get_state().get_status() != \
+		PlayAssetPackManager.AssetPackStatus.UNKNOWN
+	
+	if should_update_ui:
+		# update the downloading UI
+		var bytes_downloaded = float(request_obj.get_state().get_bytes_downloaded())
+		var bytes_to_download = float(request_obj.get_state().get_total_bytes_to_download())
+		var progress_percent = bytes_downloaded / bytes_to_download
+		var downloaded_megabyte = str(_convert_byte_to_stepified_megabytes(bytes_downloaded))
+		var total_megabyte = str(_convert_byte_to_stepified_megabytes(bytes_to_download))
+		
+		if request_obj.get_state().get_status() != PlayAssetPackManager.AssetPackStatus.COMPLETED:
+			$ProgressText.text = downloaded_megabyte + "MB/" + total_megabyte + "MB"
+			# start progress bar animation
+			$Tween.interpolate_property($ProgressBar, "value", $ProgressBar.value, \
+				int(progress_percent*100), 0.4, Tween.TRANS_QUART, Tween.EASE_OUT)
+			$Tween.start()
+
+func _reset_download_ui():
+	$Tween.interpolate_property($ProgressBar, "value", $ProgressBar.value, 0, 0.01, Tween.TRANS_QUART, Tween.EASE_OUT)
+	$Tween.start()
+	$ProgressText.text = "-MB/-MB"
+	
+func _on_cancel_button_pressed():
+	_hide_downloading_ui()
+	_show_init_ui()
+	pad_manager.cancel_asset_pack_request(pack_name)
+
+func _on_retry_button_pressed():
+	# clear previous download value
+	request_obj = pad_manager.fetch_asset_pack(pack_name)
+	request_obj.connect("request_completed", self, "handle_request_completed")
+	_reset_download_ui()
+	_hide_faileded_ui()
+	_show_downloading_ui()
+
+func _on_download_button_pressed():
+	# clear previous download value
+	request_obj = pad_manager.fetch_asset_pack(pack_name)
+	request_obj.connect("request_completed", self, "handle_request_completed")
+	_reset_download_ui()
+	_hide_init_ui()
+	_show_downloading_ui()
+
+func _on_delete_button_pressed():
+	_hide_completed_ui()
+	_show_init_ui()
+	pad_manager.remove_pack(pack_name)
+	# update downloaded_packs var in Main.gd
+	self.get_parent().get_parent().remove_downloaded_pack(pack_name)
+
+func _show_init_ui():
+	$TerminalStateText.text = "Not Installed"
+	$TerminalStateText.show()
+	$DownloadButton.show()
+
+func _hide_init_ui():
+	$TerminalStateText.hide()
+	$DownloadButton.hide()
+
+func _show_downloading_ui():
+	$ProgressBar.show()
+	$ProgressText.show()
+	$CancelButton.show()
+
+func _hide_downloading_ui():
+	$ProgressBar.hide()
+	$ProgressText.hide()
+	$CancelButton.hide()
+
+func _show_completed_ui():
+	# added extra spaces to account for the checkmark icon
+	$TerminalStateText.text = "    Download Completed"
```

Comment:
I just looked into anchoring in Godot, and it only supports anchoring to parent element or the viewport. 
Also in current implementation, we might need to not only anchor this label to CompletedIcon for ```_show_completed_ui()```, but also anchor this label to ErrorIcon for ```_show_failed_ui()```.

Right now I'm thinking about another solution, which is to use a separate label to display
```
	$TerminalStateText.text = "Not Installed"
```
Does this sounds like a good approach?

Reply:
Not answering for Jake, but could say "# added extra spaces to account for the checkmark icon because <reason>" and give a short version of the reason you commented on here.